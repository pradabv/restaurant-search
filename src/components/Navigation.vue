<template>
   <div>
    <h3>{{business.length}} {{business.length >1  ? 'Restaurants' : 'Restaurant'}}</h3>
      <div class="restaurant-list">
        <ul>
            <li @click="selectItem(item)" v-for="(item, index) in business" :key="index" :class=" selectedItem && selectedItem.id === item.id ? 'active' : ''">
                <a>
                    <h2>{{item.name}}</h2>
                    <h4 class="location">
                        <img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location">
                        <span class="address" v-if="item.location.address1">{{item.location.address1}},</span>
                        <span v-if="item.location.city">{{item.location.city}}</span>
                    </h4>
                </a>
            </li>
        </ul>
      </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import store from '@/store';
import { AppTypes } from '@/types/apptypes';

@Component
export default class Navigation extends Vue {
  get business(): AppTypes.Business[] {
    return store.state.business;
  }

  get selectedItem() {
    return store.state.selectedItem;
  }

  selectItem(item: AppTypes.Business) {
    store.commit('setSelectedItem', item);
  }
  
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.main-left h3{
    background-color: #d3d3d3;
    font-size: 16px;
    padding: 15px 30px;
}
.restaurant-list{
    height: 74vh;
    overflow-y: scroll;
    overflow-x: hidden;
}

.restaurant-list h2{
    font-size: 16px;
    margin-bottom: 10px;
}
h4.location{
    font-size: 15px;
    font-weight: normal;
    flex-direction: row;
    box-sizing: border-box;
    display: flex;
    place-content: flex-start;
    align-items: flex-start;
}
h4.location img{
    width: 20px; 
    height: auto;
    margin-right: 5px;
}
.restaurant-list ul{
    margin: 0px; 
    padding: 0px;
}
.restaurant-list ul li{
    list-style: none;
    padding: 24px 30px;
    border-bottom:#d3d3d3 1px solid;
}
.restaurant-list ul li.active{
    background-color: #61dafb;
}
.restaurant-list ul li:hover{
    background-color: #e8faff;
}
.restaurant-list ul li.active:hover{
    background-color: #61dafb;
}
.restaurant-list ul li a{
    color: #282c34;
}
.restaurant-list ul li a:hover{
    color: #282c34;
}
.location .address{
  margin-right: 5px;
}
</style>
