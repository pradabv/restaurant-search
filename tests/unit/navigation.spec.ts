
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Navigation from '@/components/Navigation.vue';
import Vuex from 'vuex';
import {sampleData} from '../mockData/mock.data';
import Store from './../../src/store/index';

const localVueWithVuex = createLocalVue();
localVueWithVuex.use(Vuex);

describe('Navigation.vue', () => {

  it('Navigation renders no records before set data in store', () => {
    const wrapper = shallowMount(Navigation, {
      localVue: localVueWithVuex
    });
    expect(wrapper.findAll('ul li').length > 0).toBeFalsy();
  });

  it('Navigation renders records after set data in store', () => {
    Store.commit('setJsonData', sampleData);
    const wrapper = shallowMount(Navigation, {
      localVue: localVueWithVuex
    });
    expect(wrapper.findAll('ul li').length).toBeGreaterThan(0);
  });

  it('Navigation trigger setSelectedItem after li click', () => {
    Store.commit('setJsonData', sampleData);
    const wrapper = shallowMount(Navigation, {
      localVue: localVueWithVuex
    });
    wrapper.find('ul li').trigger('click');
    expect(wrapper.vm.$emit('setSelectedItem')).toBeTruthy();
  });

  
  //Snapshots

  it('Navigation before selecting item MatchSnapshot', () => {
    Store.commit('setJsonData', sampleData);
    const wrapper = shallowMount(Navigation, {
      localVue: localVueWithVuex
    });
    expect(wrapper.html()).toMatchSnapshot();
  });


  it('Navigation after selecting item MatchSnapshot', () => {
    Store.commit('setJsonData', sampleData);
    const wrapper = shallowMount(Navigation, {
      localVue: localVueWithVuex
    });
    wrapper.find('ul li').trigger('click');
    expect(wrapper.html()).toMatchSnapshot();
  });

  });
  