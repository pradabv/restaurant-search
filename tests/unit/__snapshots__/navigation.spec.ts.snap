// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation.vue Navigation after selecting item MatchSnapshot 1`] = `
<div>
  <h3>1 Restaurant</h3>
  <div class="restaurant-list">
    <ul>
      <li class="active"><a>
          <h2>Billy The Butcher</h2>
          <h4 class="location"><img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location"> <span class="address">Jungfernstieg 16 - 20,</span> <span>Hamburg</span></h4>
        </a></li>
    </ul>
  </div>
</div>
`;

exports[`Navigation.vue Navigation before selecting item MatchSnapshot 1`] = `
<div>
  <h3>1 Restaurant</h3>
  <div class="restaurant-list">
    <ul>
      <li class="active"><a>
          <h2>Billy The Butcher</h2>
          <h4 class="location"><img src="../assets/icons/marker.svg" alt="restaurant location" title="restaurant location"> <span class="address">Jungfernstieg 16 - 20,</span> <span>Hamburg</span></h4>
        </a></li>
    </ul>
  </div>
</div>
`;
